### ISSUE TITLE
[Feature Request] Support for eslint.config.js (.eslintrc is being deprecated!)

### ISSUE THREAD

**Issue by aman-mehta-sf** [forcedotcom/code-analyzer#1329]
**Is your feature request related to an issue that you encountered with Salesforce Code Analyzer?**
Yes.

**Describe the solution that you want:**
.eslintrc.json is being deprecated and we would like to use the eslint.config.js. 

[Old Configuration (.eslintrc.json)](https://eslint.org/docs/latest/use/configure/configuration-files)

[New Configuration (eslint.config.js](https://eslint.org/docs/latest/use/configure/configuration-files-new)

**Describe alternatives that you've considered:**
n/a

**Additional Context:**
It would be a good idea to integrate this sooner rather than later. This would future proof our implementation of code analyzer.

**Workaround:**
Use the older format for now.

**Urgency:**
Highly Beneficial, required once eslint is upgraded to 9.x.x.

**Comment by stephen-carter-at-sf** [comment]
Yes supporting flat configuration files is on our radar, and hopefully we'll be able to schedule some time to make this work with Code Anaylzer v5 in the coming months.

**Comment by DougMidgley** [comment]
Any news on ESLINT 9 support? https://github.com/salesforce/eslint-config-lwc has been delivered, so would be good to get it added here given ESLINT 8 is no longer officially supported.

**Comment by stephen-carter-at-sf** [comment]
@DougMidgley @szymon-halik I cannot officially give you a date - there are some technical challenges we are working through to support both legacy and flat configs. I created a proposal this week for how we can support the flat config space and upgrade to ESLint v9... and currently it is internally getting reviewed. If all goes well with the proposal then we can get this work going very soon. I personally want this work done in the next month or two... and I'm trying to make it happen. But we are a small team with a lot of competing priorities.

Feel free to monitor the changes that we'll be pushing over to https://github.com/forcedotcom/code-analyzer-core/tree/dev/packages/code-analyzer-eslint-engine in weeks to come.
